src/compiler/scala/tools/nsc/Reporting.scala

```diff
@@ -93,12 +93,12 @@ trait Reporting extends internal.Reporting { self: ast.Positions with Compilatio
       else false
 
     private def registerErrorTextEdit(pos: Position, msg: String, actions: List[CodeAction]): Boolean = {
-      val matches = quickfixFilters.exists({
-        case MessageFilter.Any => true
+      val matches = quickfixFilters.exists {
+        case MessageFilter.Any => actions.exists(_.edits.nonEmpty)
         case mp: MessageFilter.MessagePattern => mp.check(msg)
```

Comment:
Also for other filters:

```
➜ sandbox sc Test.scala '-quickfix:msg=type mismatch'
Test.scala:2: error: [rewritten by -quickfix] type mismatch;
 found   : Int(1)
 required: String
  def f: String = 1
                  ^
1 error
```

Reply:
Thanks, I'll follow up in our morning / post-sunrise.

Reply:
Or I better do it now before I fall asleep and forget everything in dreams.